gems/pending/middleware/events/middleware_event_monitor.rb

```diff
@@ -0,0 +1,31 @@
+class MiddlewareEventMonitor
+  def initialize(ems, start_time)
+    @ems = ems
+    @alerts_client = ems.alerts_connect
+    # TODO: can we store and resume where we left off?  To avoid duplicates, only look for events now or later,
+    # so basically we miss events when we're not running...
+    @start_time = start_time || Time.now.to_i * 1000
+  end
+
+  def start
+    # nothing to do, not an active connection, just making rest calls
+  end
+
+  def stop
+    # nothing to do, not an active connection, just making rest calls
+  end
+
+  def each
+    # Just fetch up to the current time, then rest current time as the next start time
+    end_time = Time.now.to_i * 1000
+    $mw_log.debug("Catching Events for interval [Start=#{@start_time}, End=#{end_time}]")
+
+    @alerts_client.list_events("startTime" => @start_time, "endTime" => end_time).each do |event|
+      yield event
```

Comment:
isnt it used [here?](https://github.com/ManageIQ/manageiq/pull/7395/files/a526f83b6155051f10c95a24727f6869134d37ef#diff-7de3a2b96e91ea35242b90bfd597d1a4R32
)

I think monitor is misleading too, because it doesnt monitor yet, but fetches the events only once.
Have a look at @bronaghs PR for that. I think you can copy her pattern right away (which also went through some iterations already :)